dbt-snowflake/src/dbt/include/snowflake/macros/relations/dynamic_table/create.sql

```diff
@@ -1,85 +1,14 @@
 {% macro snowflake__get_create_dynamic_table_as_sql(relation, sql) -%}
-{#-
---  Produce DDL that creates a dynamic table
---
---  Args:
---  - relation: Union[SnowflakeRelation, str]
---      - SnowflakeRelation - required for relation.render()
---      - str - is already the rendered relation name
---  - sql: str - the code defining the model
---  Globals:
---  - config: NodeConfig - contains the attribution required to produce a SnowflakeDynamicTableConfig
---  Returns:
---      A valid DDL statement which will result in a new dynamic table.
--#}
 
+    {%- set catalog_relation = adapter.build_catalog_relation(config.model) -%}
     {%- set dynamic_table = relation.from_config(config.model) -%}
 
-    {%- if dynamic_table.catalog.table_format == 'iceberg' -%}
-        {{ _get_create_dynamic_iceberg_table_as_sql(dynamic_table, relation, sql) }}
+    {%- if catalog_relation.catalog_type == 'NATIVE' -%}
+        {{ snowflake__create_dynamic_table_standard_sql(dynamic_table, relation, compiled_code) }}
+    {%- elif catalog_relation.catalog_type == 'ICEBERG_MANAGED' -%}
+        {{ snowflake__create_dynamic_table_iceberg_managed_sql(dynamic_table, relation, compiled_code) }}
     {%- else -%}
-        {{ _get_create_dynamic_standard_table_as_sql(dynamic_table, relation, sql) }}
+        {% do exceptions.raise_compiler_error('Unexpected model config for: ' ~ relation) %}
     {%- endif -%}
 
 {%- endmacro %}
-
-
-{% macro _get_create_dynamic_standard_table_as_sql(dynamic_table, relation, sql) -%}
-{#-
---  Produce DDL that creates a standard dynamic table
---
---  This follows the syntax outlined here:
---  https://docs.snowflake.com/en/sql-reference/sql/create-dynamic-table#syntax
---
---  Args:
---  - dynamic_table: SnowflakeDynamicTableConfig - contains all of the configuration for the dynamic table
---  - relation: Union[SnowflakeRelation, str]
---      - SnowflakeRelation - required for relation.render()
---      - str - is already the rendered relation name
---  - sql: str - the code defining the model
---  Returns:
---      A valid DDL statement which will result in a new dynamic standard table.
--#}
-
-    create dynamic table {{ relation }}
-        target_lag = '{{ dynamic_table.target_lag }}'
-        warehouse = {{ dynamic_table.snowflake_warehouse }}
-        {{ optional('refresh_mode', dynamic_table.refresh_mode) }}
-        {{ optional('initialize', dynamic_table.initialize) }}
-        as (
-            {{ sql }}
-        )
-
-{%- endmacro %}
-
-
-{% macro _get_create_dynamic_iceberg_table_as_sql(dynamic_table, relation, sql) -%}
```

Comment:
Should we keep this macro name as is? Can we extend the templating later to handle non-managed catalogs as needed? Just thinking that the diff here is really just `catalog = 'SNOWFLAKE'`

Reply:
I'd like to follow a naming convention across relation types. I'm open to what that convention is, but I'd like to account for the different DDL options across different Iceberg catalog integrations with different templates, to make it clear what is and isn't supported, and avoid trying to set a setting that does not apply to a specific catalog integration.

Reply:
maybe the relation/format level macro like  `_get_create_dynamic_iceberg_table_as_sql` encapsulates the switching logic to call the correct catalog templating macro?